### ISSUE TITLE
Missing explicit dependency and include on log4cxx

### ISSUE THREAD

**Issue by dirk-thomas** [ros-drivers/openni_camera#21]
In the latest version of rosconsole the strict dependency on log4cxx has been removed. Now `openni_camera` (http://jenkins.ros.org/view/HbinP32/job/ros-hydro-openni-camera_binarydeb_precise_i386/27/console) fails to build since it relied on the transitive dependency.

Please make the dependency on log4cxx explicit and import the necessary header file.

**Comment by piyushk** [comment]
Solution pending discussion on ros-drivers/hokuyo_node#8

**Comment by dirk-thomas** [comment]
Please address the missing explicit dependency on log4cxx to unbreak the build on the farm.

**Comment by tfoote** [comment]
Fixed in 00c71c3

**Comment by jack-oquin** [comment]
I don't think the fix above is complete, yet. 

It is hard to test it on my own system which has the released version of ros_comm, containing this in `ros/console.h`:

``` C++
// TODO: this header is no longer needed to be included here, but removing it will break various code that incorrectly does not itself include log4cxx/logger.h
// We should vet all the code using log4cxx directly and make sure the includes/link flags are used in those packages, and then we can remove this include
#include <log4cxx/logger.h>
```

For years, that had provided log4cxx symbols to the namespace of _every_ program containing `#include <ros/ros.h>`.

Making that change in the "stable" Hydro release was a really bad idea. We should have followed the advice of whoever wrote that comment.

**Comment by tfoote** [comment]
Fixed in 9914fe09efcb2140efd8e653a04f3aae8e93351b and building on the farm.

**Comment by bingxiangniu** [comment]
sorry
